mods/bones/init.lua

```diff
@@ -23,24 +23,38 @@ local function is_owner(pos, name)
 	return false
 end
 
+local function appendmulti(tbl,...)
+  for _, v in pairs({...}) do
+	table.insert(tbl, v)
+  end
+end
+
 local function get_bones_formspec_for_size(numitems)
 	local cols, rows
+	local scroll=false
 	if numitems <= min_inv_size then
 		cols, rows = 8, 4
-	elseif numitems <= 4 * 15 then
-		cols, rows = math.ceil(numitems / 4), 4
 	else
-		cols, rows = 15, math.ceil(numitems / 15)
+		cols, rows = 8, math.ceil(numitems / 8)
+		scroll=true
+	end
+	local output={}
+	appendmulti(output, "size[", 8.5, ",", 9, "]")
+	if scroll  then
+		appendmulti(output, "scrollbaroptions[max=",rows*9.3,"]")
```

Comment:
the problem with the scrollbar might be lying here, maybe the proper formula would be something like (rows-delta)*multiplier

Reply:
A `max=` value directly expressed in inventory slots would be even better. I added it to my long term notes to have a look at in the Minetest Engine.

Reply:
well, either  inventory slots or inventory rows but it would be easier to use, I agree